nilearn/plotting/img_plotting.py

```diff
@@ -67,12 +67,8 @@ def _plot_img_with_bg(img, bg_img=None, cut_coords=None,
         data = img.get_data()
         affine = img.get_affine()
 
-        # Remove NaNs
-        nan_mask = np.isnan(np.asarray(data))
-        if np.any(nan_mask):
-            data = data.copy()
-            data[nan_mask] = 0
-        del nan_mask
+        if np.any(np.isnan(data)):
```

Comment:
An additional tiny remark there is a slight difference vs the previous version of the code because you are not using np.asarray any more.

This can make a difference for masked or memapped arrays. Whether this actually matters in practice I am not sure.